### PR TITLE
Use default JWT when unauthenticated

### PR DESCRIPTION
**Pull request by tfussell**
Using a special unauthenticated JWT for requests when a user is not logged in. This goes along with this apiserver PR https://github.com/venturemark/apiserver/pull/277 It means istio can handle all auth concerns while the apiserver can assume that all requests are authenticated.

### CODE REVIEW

File: src/module/apikeys/index.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
+export const DefaultToken =
+  "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImlzcyI6InB1YmxpYy52ZW50dXJlbWFyay5jbyJ9.eyJpc3MiOiJwdWJsaWMudmVudHVyZW1hcmsuY28iLCJzdWIiOiJ1bmF1dGhlbnRpY2F0ZWQiLCJhdWQiOlsiYXBpc2VydmVyIl19.C653WdHfPiKvAPQkEFYlBuDCHzDvsZY6XQmp768OdiZom5fi_EMQubiS6Of_M-n4exTdk4-ZsXqTA6hhzN8rW6ManHDftB8Z_kCkJFqdN2CP2YJA3RDz5w4U3rYVm4E0r8rY5BLX1NLytrBaFRQGm3yNoc-Y_KYV-2UTNN6D1DUFov9kIMMmLHF1FU2XjVtqbA8ApRXCm5HpYCSMUVyf2MGlVP-qxaEDrpUpun6wHVk_myQmQYzBqRPm0AAA2TUQ3ForjM6OPzlehg18tvMgMZRdTObRSEKlEpsj_D8Johe2b7kMX4JQsv0_x6VdIcePAMsAkT7we1MfIvajY6h74A";
```
Comment (tfussell) [inline src/module/apikeys/index.ts:2]: Generated following https://github.com/venturemark/jwks

File: src/module/api/venture/search/index.ts [hunk 2]
```diff
@@ -25,8 +25,6 @@ export async function Search(
   ventureSearch.id && obj.getMetadataMap().set(key.VentureID, ventureSearch.id);
   ventureSearch.userId &&
     obj.getMetadataMap().set(key.SubjectID, ventureSearch.userId);
-  ventureSearch.slug &&
-    obj.getMetadataMap().set("venture.venturemark.co/slug", ventureSearch.slug);
```
Comment (tfussell) [inline src/module/api/venture/search/index.ts:29]: Functionality no longer needed.

File: src/module/api/timeline/update/index.ts [hunk 3]
```diff
@@ -55,14 +51,22 @@ export async function Update(
...
-  if (updateTimeline.membersWrite !== undefined) {
+  if (updateTimeline.membersWrite) {
     const patch = new UpdateI_Obj_Jsnpatch();
     patch.setOpe("replace");
     patch.setPat(`/obj/metadata/${key.PermissionModel.replace("/", "~1")}`);
     patch.setVal(updateTimeline.membersWrite ? "writer" : "reader");
     patchList.push(patch);
   }
 
+  if (updateTimeline.visibility) {
```
Comment (tfussell) [inline src/module/api/timeline/update/index.ts:62]: Unrelated, allows editing timeline visibility.

File: src/module/api/timeline/update/index.ts [hunk 4]
```diff
@@ -25,10 +25,6 @@ export async function Update(
   obj.getMetadataMap().set(key.TimelineID, updateTimeline.id);
   obj.getMetadataMap().set(key.VentureID, updateTimeline.ventureId);
 
-  if (updateTimeline.visibility) {
-    obj.getMetadataMap().set(key.ResourceVisibility, updateTimeline.visibility);
-  }
-
```
Comment (peterphan1996) [inline src/module/api/timeline/update/index.ts:31]: What is the different between this one and the one you changed to 
```
if (updateTimeline.visibility) {
     const patch = new UpdateI_Obj_Jsnpatch();
     patch.setOpe("replace");
     patch.setPat(`/obj/metadata/${key.ResourceVisibility.replace("/", "~1")}`);
     patch.setVal(updateTimeline.visibility);
     patchList.push(patch);
   }
```

Comment (tfussell) [reply src/module/api/timeline/update/index.ts:31]: `obj.getMetadataMap()` sets the metadata for the API request (sort of like HTTP headers) while the JSON patch is a patch to be applied to the actual object in the database. These are operating at different levels of responsibility--GRPC layer vs business layer. Sometimes this is confusing though which is why I made this mistake :)

**Pull request merged**